events.cpp

```diff
@@ -28,13 +28,14 @@
 #include <map>
 #include <algorithm>
 #include "gamedata.h"
+#include <sstream>
 
 FactBase::~FactBase() {
 
 }
 
 void BattleSide::AssignUnit(Unit* unit) {
-	this->factionName = unit->faction->name->Str();
+	this->factionName = unit->faction->name().Str();
```

Comment:
why did it change?

Reply:
For JSON, I print the faction name and number separately (currently, just in the faction status, but it could be done everywehre). The old name field had both, now name_ member just as the raw name, and name() method returns the name and number.